### DISCUSSION TITLE
The Selenium testing example "hello_tauri" is returning error on Windows

### DISCUSSION THREAD

**Discussion by marcelsby** [tauri-apps/tauri#10122] [Q&A]
Hello, im trying to e2e test my Tauri app and to start learning how to do it i call upon the documentation. Actually im on Windows 11, then i've done the following steps:

1. Checked my Edge browser version and downloaded the right `msedgedriver.exe` version, then included it in `$PATH`.
2. Installed the `tauri-driver` with the `cargo install tauri-driver` command.
3. Cloned the "hello_tauri" from [here](https://github.com/chippers/hello_tauri).
4. Build and run the app with `cargo run --release`, for the first time.
5. Tried to run the tests in the `webdriver/selenium` directory, but received the following error:

![image](https://github.com/tauri-apps/tauri/assets/57691142/28602ef8-d6c1-4e6a-81a6-10b4b0a81580)

I've tried to set the `ms:edgeOptions` capability to a empty object but i receive the same error, then i attempted to set the Microsoft Edge driver options with help of [Selenium docs](https://github.com/SeleniumHQ/seleniumhq.github.io/blob/trunk//examples/javascript/test/getting_started/openEdgeTest.spec.js#L11-L15), but the same error again. Any clue how to solve this?

The test file:

```js
const os = require("os");
const path = require("path");
const { expect } = require("chai");
const { spawn, spawnSync } = require("child_process");
const { Builder, By, Capabilities } = require("selenium-webdriver");

// create the path to the expected application binary
const application = path.resolve(
  __dirname,
  "..",
  "..",
  "..",
  "target",
  "release",
  "hello_tauri"
);

// keep track of the webdriver instance we create
let driver;

// keep track of the tauri-driver process we start
let tauriDriver;

before(async function() {
  // set timeout to 2 minutes to allow the program to build if it needs to
  this.timeout(120000)

  // ensure the program has been built
  spawnSync("cargo", ["build", "--release"]);

  // start tauri-driver
  tauriDriver = spawn(
    path.resolve(os.homedir(), ".cargo", "bin", "tauri-driver"),
    [],
    { stdio: [null, process.stdout, process.stderr] }
  );

  const capabilities = new Capabilities();
  capabilities.set("tauri:options", { application });
  capabilities.setBrowserName("wry");

  // start the webdriver client
  driver = await new Builder()
    .withCapabilities(capabilities)
    .usingServer("http://localhost:4444/")
    .build();
});

after(async function() {
  // stop the webdriver session
  await driver.quit();

  // kill the tauri-driver process
  tauriDriver.kill();
});

describe("Hello Tauri", () => {
  it("should be cordial", async () => {
    const text = await driver.findElement(By.css("body > h1")).getText();
    expect(text).to.match(/^[hH]ello/);
  });

  it("should be excited", async () => {
    const text = await driver.findElement(By.css("body > h1")).getText();
    expect(text).to.match(/!$/);
  });

  it("should be easy on the eyes", async () => {
    // selenium returns color css values as rgb(r, g, b)
    const text = await driver.findElement(By.css("body")).getCssValue("background-color");

    const rgb = text.match(/^rgb\((?<r>\d+), (?<g>\d+), (?<b>\d+)\)$/).groups;
    expect(rgb).to.have.all.keys('r','g','b');

    const luma =  0.2126 * rgb.r + 0.7152 * rgb.g  + 0.0722 * rgb.b ;
    expect(luma).to.be.lessThan(100)
  });
});
```

**Comment by Nickog13** [comment]
Hi
Just had the same issue in my tauri app Selenium automation after upgrading to the latest tauri-driver version.
You should add the webviewOptions to the options of tauri like that:

    capabilities.set('tauri:options', {
        application: application,
        webviewOptions: {},
    });
    capabilities.setBrowserName('wry')

This fixed the issue for me.
